### PR TITLE
enh : Add regression to the mean model

### PR DESCRIPTION
**Pull request by celprov**
Naive implementation of the AverageDWModel proposed in the nipreps-book. #41

### CODE REVIEW

**Review by arokem — COMMENTED** [review]
One small comment from me: the name `AverageDWModel` suggests that this is a model that averages the diffusion-weighted signals, but in practice, this averages both diffusion-weighted *and* non-diffusion-weighted (i.e., b0) signals.

One related suggestion is to give this model a bit more flexibility by allowing some arguments that will control which b-values are used in averaging (for example, `lb` and `ub`  keyword arguments that specify a range)

**Comment by oesteban** [comment]
> One small comment from me: the name `AverageDWModel` suggests that this is a model that averages the diffusion-weighted signals, but in practice, this averages both diffusion-weighted _and_ non-diffusion-weighted (i.e., b0) signals.

In principle, this should only receive DW volumes (i.e., no b=0). However, I admit that I created a little loophole for it here: https://github.com/nipreps/eddymotion/blob/c7de9f7c4d687c7064bc321fad829479b0730af6/eddymotion/dmri.py#L50

Effectively, if you generated the training set with `with_b0=True`, you'd have a b=0 at the beginning of the dataset.

Because checking the gradient table to see if there are or aren't b=0 volumes is almost free, I'll make a quick suggestion for that.

> One related suggestion is to give this model a bit more flexibility by allowing some arguments that will control which b-values are used in averaging (for example, `lb` and `ub` keyword arguments that specify a range)

Yes, I had already discussed this with @celprov, and I warned her that I would give this a thorough review including suggestions (including this one among others). The idea was to create a base of a PR as an exercise and build from there, given that she is starting with GitHub, coding, etc.

File: eddymotion/model.py [hunk 1]
```diff
@@ -91,6 +91,24 @@ def predict(self, gradient, **kwargs):
         return self._S0
 
 
+class AverageDWModel:
+    """A trivial model that returns an average map."""
+
+    __slots__ = ("_data",)
```
Comment (oesteban) [inline eddymotion/model.py:97]:
```suggestion
    __slots__ = ("_data", "_gtab")
```

File: eddymotion/model.py [hunk 2]
```diff
@@ -91,6 +91,24 @@ def predict(self, gradient, **kwargs):
...
+
+    def __init__(self, gtab, **kwargs):
+        """Implement object initialization."""
+        return  # do nothing at initialization time
```
Comment (oesteban) [inline eddymotion/model.py:101]:
```suggestion
        self._gtab = gtab
```

File: eddymotion/model.py [hunk 3]
```diff
@@ -91,6 +91,24 @@ def predict(self, gradient, **kwargs):
...
+
+    def fit(self, data, **kwargs):
+        """Calculate the average."""
+        self._data = data.mean(-1)
```
Comment (oesteban) [inline eddymotion/model.py:105]:
```suggestion
        self._data = data[..., self._gtab[..., 3] > 50].mean(-1)
```

File: eddymotion/model.py (hunk 2 above)
Comment (oesteban) [reply eddymotion/model.py:101]: keep a reference to the gradient table, so that later you can check on it when creating the average.

**Review by oesteban — APPROVED** [review]
Looking good! Added some suggestions addressing @arokem's concern. Thanks much for this @celprov !

**Pull request merged**